drivers/sensor/am2320/am2320.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) https://github.com/tidyjiang8
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <i2c.h>
+#include <init.h>
+#include <misc/__assert.h>
+#include <misc/byteorder.h>
+#include <sensor.h>
+#include <string.h>
+#include <crc16.h>
+#include <logging/sys_log.h>
+
+/* am2320 i2c address, 7+1 bit */
+#define AM2320_I2C_ADDR     0xB8
+
+struct am2320_data {
+	struct device *i2c;
+	s16_t temp;
+	s16_t humidity;
+};
+
+static u16_t am2320_crc16(u8_t *ptr, unsigned int len)
+{
+	u16_t   crc = 0xFFFF;
+	u8_t    i;
+
+	while (len--) {
+		crc ^= *ptr++;
+
+		for (i = 0; i < 8; i++) {
+			if (crc & 0x01) {
+				crc >>= 1;
+				crc ^= 0xA001;
+			} else {
+				crc >>= 1;
+			}
+		}
+	}
+
+	return crc;
+}
+
+static int am2320_channel_get(struct device *dev,
+					enum sensor_channel chan,
+					struct sensor_value *val)
+{
+	struct am2320_data *drv_data = dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_TEMP || SENSOR_CHAN_HUMIDITY);
+
+	if (chan == SENSOR_CHAN_TEMP) {
+		val->val1 = drv_data->temp / 10;
+		val->val2 = drv_data->temp % 10;
+	} else { /* SENSOR_CHAN_HUMIDITY */
+		val->val1 = drv_data->humidity / 10;
+		val->val2 = drv_data->humidity % 10;
+	}
+
+	return 0;
+}
+
+static int am2320_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct am2320_data *drv_data = dev->driver_data;
+	u8_t   buf[8];
+	u16_t  crc;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* Step 0: clean stored old data. */
+	drv_data->temp     = 0;
+	drv_data->humidity = 0;
+
+	/**
+	 * Step 1: Wake up AM2320.
+	 * Send START + ADDRESS(0xA8) signal to wake up AM2320.
+	 * We do not check the return value, as AM2320 will not
+	 * response ACK signal.
+	 */
+	buf[0] = AM2320_I2C_ADDR;
+	i2c_write(drv_data->i2c, buf, 1, AM2320_I2C_ADDR >> 1);
```

Comment:
You are right, the datasheet does state that you need to explicitly wake up the chip fist. Initially I though that it was done by the implicit address sent by the i2c_write.

So you can leave the way you had in the first version. Sorry for the confusion.